### ISSUE TITLE
Support self-signed / insecure ssl certficates

### ISSUE THREAD

**Issue by freshteapot** [varnish/vagent2#182]
Currently it is not possible to run the agent against a VAC (etc) running a self-signed certificate or an insecure ssl.

We get an error in the logs warning of:

```
Jan 18 15:37:37 cache1 varnish-agent[5402]: send_curl (modules/vac_register.c:80): Sending VAC registration request: https://192.168.50.22/api/rest/register
Jan 18 15:37:37 cache1 varnish-agent[5402]: issue_curl (modules/curl.c:108): Issuing curl command with url=https://192.168.50.22/api/rest/register. Request body present
Jan 18 15:37:38 cache1 varnish-agent[5402]: issue_curl (modules/curl.c:133): Curl callback failed with status code 60
```

Inside [curl.c](https://github.com/varnish/vagent2/blob/master/src/modules/curl.c#L119), it would be nice if we could parse in a parameter at the command line like you can do in curl via '-k' to take advantage of CURLOPT_SSL_VERIFYPEER and set it to 0.

```
curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, 0);
```

Running it via the curl command to confirm the issue is the same.

```
curl -i -v 'https://192.168.50.22/api/rest/register?hostname=192.168.50.22'
* Hostname was NOT found in DNS cache
*   Trying 192.168.50.22...
* Connected to 192.168.50.22 (192.168.50.22) port 443 (#0)
* successfully set certificate verify locations:
*   CAfile: none
  CApath: /etc/ssl/certs
* SSLv3, TLS handshake, Client hello (1):
* SSLv3, TLS handshake, Server hello (2):
* SSLv3, TLS handshake, CERT (11):
* SSLv3, TLS alert, Server hello (2):
* SSL certificate problem: self signed certificate
* Closing connection 0
curl: (60) SSL certificate problem: self signed certificate
More details here: http://curl.haxx.se/docs/sslcerts.html
```

## Reference
- http://curl.haxx.se/docs/sslcerts.html
- curl [error 60](https://curl.haxx.se/libcurl/c/libcurl-errors.html#CURLESSLCACERT)

**Comment by Dridi** [comment]
Thanks!

Don't forget to add `Fixes #182` in your commit message to automatically close tickets when pull request are merged.

**Comment by freshteapot** [comment]
How exciting!

Fixes: I have been wondering, as I used to do it, but lately I have noticed I have been writing it wrong. So thank you :)
